src/main/java/org/threadly/concurrent/PriorityScheduler.java

```diff
@@ -456,37 +358,12 @@ public void prestartAllCoreThreads() {
    * @param value {@code true} if core threads should be expired when idle.
    */
   public void allowCoreThreadTimeOut(boolean value) {
-    boolean checkForExpiredWorkers = ! allowCorePoolTimeout && value;
-    
-    allowCorePoolTimeout = value;
-    
-    if (checkForExpiredWorkers) {
-      synchronized (workersLock) {
-        expireOldWorkers();
-      }
-    }
+    workerPool.allowCoreThreadTimeOut(value);
   }
 
   @Override
   public boolean isShutdown() {
-    return shutdownStarted.get();
-  }
-  
-  /**
-   * Stops all idle workers, this is expected to be part of the shutdown process.
-   */
-  protected void shutdownAllIdleWorkers() {
```

Comment:
Functions like this were able to be completely moved to the new inner class of "WorkerPool"